### PR TITLE
🌍 Use global normalization for KL logging (to match normalization for loss)

### LINKED ISSUE #2995: Loss normalization in GRPOTrainer
When beta = 0 and num_iterations = 1 in GRPO, the loss should be exactly 0 by doing the math. However, I'm getting non-zero loss, and I think it's related to line 950 in grpo_trainer.py:

`        loss = (per_token_loss * completion_mask).sum() / completion_mask.sum()`

In this case, we are normalizing the advantages across the entire batch of sequences (with potentially different lengths). This is a different formula from the original GRPO algorithm, where we only normalize across each individual sequence. 

I changed the line to: 
`        loss = ((per_token_loss * completion_mask).sum(dim=1) / completion_mask.sum(dim=1)).mean()`
and I got a constant loss of zero, which is correct.

### PR DESCRIPTION
**Pull request by tchang1997**
# What does this PR do?

Fixes normalization in the mean KL divergence for logging to match the global normalization of the per-token losses. 

Fixes #2995  (partial; [see comment](https://github.com/huggingface/trl/issues/2995#issuecomment-2695599136)).

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by qgallouedec — APPROVED** [review]
Thanks!
